uplink/src/base/actions/mod.rs

```diff
@@ -220,34 +216,20 @@ impl Actions {
                 self.tunshell_tx.send_async(action).await?;
                 return Ok(());
             }
+            #[cfg(target_os = "linux")]
             "configure_journalctl" => {
-                match serde_json::from_str::<JournalctlConfig>(action.payload.as_str()) {
-                    Ok(mut journalctl_config) => {
-                        journalctl_config.tags = journalctl_config.tags.into_iter()
-                            .filter(|tag| !tag.is_empty())
-                            .collect();
-                        log::info!("restarting journalctl with following config: {:?}", journalctl_config);
-                        self.journalctl_instance = Some(JournalctlInstance::new(self.create_log_stream(), &journalctl_config))
-                    }
-                    Err(e) => {
-                        error!("couldn't parse journalctl config payload:\n{}\n{}", action.payload, e)
-                    }
-                }
+                let mut config = serde_json::from_str::<LoggingConfig>(action.payload.as_str())?;
+                config.tags = config.tags.into_iter().filter(|tag| !tag.is_empty()).collect();
+                log::info!("restarting journalctl with following config: {:?}", config);
+                self.logger = Some(new_journalctl(self.create_log_stream(), &config))
```

Comment:
These two options could be merged as well.

Reply:
I feel like we need to move all of this code outside of `Actions` and this seems like a good contender for re-writing as a collector.